pkg/k8s/watchers/cilium_endpoint.go

```diff
@@ -235,3 +252,59 @@ func (k *K8sWatcher) endpointDeleted(endpoint *types.CiliumEndpoint) {
 		k.egressGatewayManager.OnDeleteEndpoint(endpoint)
 	}
 }
+
+// ciliumEndpointManager manages tasks related to ciliumendpoints, including
+// performing periodic cleanup of local stale CEP resources.
+type ciliumEndpointManager struct {
+	*lock.RWMutex
+	ciliumEndpointsCleanupEnabled bool
+	markedForGC                   []*types.CiliumEndpoint
+	client                        v2.CiliumV2Interface
+}
+
+func newCiliumEndpointManager(ciliumClient v2.CiliumV2Interface) *ciliumEndpointManager {
+	return &ciliumEndpointManager{
+		RWMutex: &lock.RWMutex{},
+		client:  ciliumClient,
+	}
+}
+
+func (cpm *ciliumEndpointManager) markForDeletion(cep *types.CiliumEndpoint) {
+	cpm.Lock()
+	defer cpm.Unlock()
+	cpm.markedForGC = append(cpm.markedForGC, cep)
+}
+
+func (cpm *ciliumEndpointManager) sweep(ctx context.Context) error {
+	cpm.Lock()
+	defer cpm.Unlock()
+	if cpm.markedForGC == nil {
+		return nil
+	}
```

Comment:
This `nil` check seems superfluous, as the loop would not be entered when the slice is `nil` (`len` of a `nil` slice is 0). This code actually depends on it already as reslicing the slice does not make it `nil` when empty?